larpandora/LArPandoraEventBuilding/LArPandoraTrackCreation_module.cc

```diff
@@ -126,15 +126,15 @@ void LArPandoraTrackCreation::produce(art::Event &evt)
 
     // ATTN: Expectations here are that the input geometry corresponds to either a single or dual phase LArTPC.  For single phase we expect
     // three views, U, V and either W or Y, for dual phase we expect two views, W and Y.
-    const bool isDualPhase(theGeometry->MaxPlanes() == 2);
+    const bool isDualPhase(LArPandoraHelper::IsDualPhase());
 
     if (nWirePlanes != planeSet.size())
         throw cet::exception("LArPandoraTrackCreation") << " LArPandoraGeometry::LoadGeometry --- geometry description for wire plane(s) missing ";
 
     if (isDualPhase && (!planeSet.count(geo::kW) || !planeSet.count(geo::kY)))
         throw cet::exception("LArPandoraTrackCreation") << " LArPandoraGeometry::LoadGeometry --- dual phase scenario; expect to find w and y views ";
 
-    if (!isDualPhase && (!planeSet.count(geo::kU) || !planeSet.count(geo::kV) || (planeSet.count(geo::kW) && planeSet.count(geo::kY))))
+    if (isDualPhase && (!planeSet.count(geo::kU) || !planeSet.count(geo::kV) || (planeSet.count(geo::kW) && planeSet.count(geo::kY))))
```

Comment:
Is this change intended? The error message appears to be assuming we're in a single phase scenario - this looks like it would throw an exception if everything was as expected for both single phase and dual phase.

As a side note, not something that has changed in this PR, but the commentary suggest the single phase case should expect U, V and W **or** Y to be set, but most of the cases here appear to assume that W **and** Y will vary together. Do we know which behaviour we can rely on?

Reply:
Thanks @AndyChappell, I was pretty sure I'd reverted this change (that was accidental!), need to look into why it's not been propagated!

Reply:
Fixed - I had just forgotten to push :)